### DISCUSSION TITLE
Consider how to make people end their time at Onramp with actual sample applications

### DISCUSSION THREAD

**Discussion by mattstauffer** [tightenco/onramp#384] [Ideas]
Possibly suggesting an app that they can build, step-by-step, as a later add-on? So each module would have a goal that your app could do, and then when you're done with Onramp you have built an entire app from scratch in Laravel

**Comment by gcavanunez** [comment]
Agree 100%, would help them on the journey of making their portafolio. Could even be something sort of like how freecodecamp does it with their projects at the end of a module where they attach a link of the codepen of the project. In this case however a github repo to check out.

**Comment by nicorafales** [comment]
I vote github repo so they can integrate GIT onto their proven skills.
Look at how SymfonyCasts has done it, where they start a project and in the first video of a series, there is a download button, which gets you a zip with two folders, _start_ and _end_ and in each folder you get the starting template (since it's a specific series) and you can work your way up through the videos on-par with what the lesson is on.

**Making them do a blog** has the advantage of being first-class supportive with code/spec examples from official laravel documentation, so instead of parsing a similar problem to what they have to solve, they can focus purely on tooling, coding, bug-fixing. They can tacle a more advanced idea after they have a base knowledge on everything.

**Comment by zuzana-kunckova** [comment]
Can I just suggest to do something else than a blog? I think there are many tutorials and websites describing how to build a blog (or a ToDo app) already. Maybe offer them a list of projects to choose from?

**Comment by Treggats** [comment]
Would I be crazy to suggest we guide them how to build their own CMS?

**Comment by atefBB** [comment]
@Treggats good idea !

**Comment by mattstauffer** [comment]
I think I most love something that includes the opportunity for some more advanced concepts--queues, S3, a little bit of Vue. I've been thinking for a long time that I might want to have them build a *private community* version of Instagram, just for their friends and family.

**Comment by zuzana-kunckova** [comment]
I really love that idea! Very interesting and definitely something people (I) would use.

**Comment by jonsugar** [comment]
This is something I'm actually super interested in. I would love to make app like that as a test of my own abilities. I wouldn't be following along though Onramp but as an exercise it would be awesome. 

If there was a spec for it that would be great. Having clear requirements would focus me. I remember the Tighten dev clash where you built the Twitter clones in Vue and React. There was a spec. involved in that.

**Comment by mattstauffer** [comment]
https://twitter.com/FrontendMasters/status/1172584067499470848

^^ example of how to do this

**Comment by mattstauffer** [comment]
@jonsugar Sorry I didn't see your message. I'll work on a spec as soon as I can :)

**Comment by Treggats** [comment]
@mattstauffer I'm peeking in the Elm community. They use https://github.com/gothinkster/realworld as a spec to build a demo app https://dev.to/rtfeldman/tour-of-an-open-source-elm-spa

**Comment by davorminchorov** [comment]
It would be a good idea if there are multiple project ideas for people to choose from to experiment with the tech stack that they just learned besides the projects they built while watching the courses (or any other type of learning resource).

I am thinking something bigger than a blog, like some kind of platform: ERP, CRM, E-Commerce, Job / Ad Board, Money management type of application that will take them a few months to build. 

This will let them learn about more advanced concepts like Queues, External APIs, REST APIs, Caching,, File Uploads, web sockets etc.

Also, people will be able to sharpen their skills that they just learned and get a job faster and be prepared for the real world.
